### PR TITLE
GH-14824: [CI] r-binary-packages should only upload artifacts if all tests succeed 

### LINKED ISSUE #14824: [CI] r-binary-packages should only upload artifacts if all tests succeed
### Describe the enhancement requested

Currently the upload step is missing a dependency on the centos binary test.

### Component(s)

Continuous Integration

### PR DESCRIPTION
**Pull request by assignUser**
* Closes: #14824

### CODE REVIEW

**Comment by assignUser** [comment]
@github-actions crossbow submit r-binary-packages

**Review by thisisnic — APPROVED** [review]
LGTM!

File: dev/tasks/r/github.packages.yml [hunk 1]
```diff
@@ -355,7 +355,7 @@ jobs:
 
   upload-binaries:
     # Only upload binaries if all tests pass.
-    needs: [ r-packages, test-source, test-linux-binary]
+    needs: [ r-packages, test-source, test-linux-binary, test-centos-binary]
```
Comment (kou) [inline dev/tasks/r/github.packages.yml:358]:
```suggestion
    needs: [r-packages, test-source, test-linux-binary, test-centos-binary]
```

**Review by kou — APPROVED** [review]
+1

**Pull request merged**